ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/streams/QUICStreams.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.quic.streams
+
+import io.ktor.network.sockets.*
+import kotlinx.coroutines.channels.*
+
+public typealias QUICStream = ByteArray
+
+/**
+ * A channel for sending QUIC streams
+ */
+public interface QUICStreamWriteChannel {
+    /**
+     * QUIC stream outgoing channel.
+     */
+    public val outgoing: SendChannel<QUICStream>
+
+    /**
+     * Send QUIC stream.
+     */
+    public suspend fun send(stream: QUICStream) {
+        outgoing.send(stream)
+    }
+}
+
+/**
+ * A channel for receiving QUIC streams
+ */
+public interface QUICStreamReadChannel {
+    /**
+     * Incoming QUIC streams channel
+     */
+    public val incoming: ReceiveChannel<QUICStream>
+
+    /**
+     * Receive a QUIC stream.
+     */
+    public suspend fun receive(): QUICStream = incoming.receive()
+}
+
+/**
+ * A channel for sending and receiving QUIC streams
+ */
+public interface QUICStreamReadWriteChannel : QUICStreamWriteChannel, QUICStreamReadChannel
```

Comment:
Can be internal

Reply:
These interfaces are exposed via `aScoket().quic()`, do we want to hide it and allow only `HTTP/3` to use `QUIC`?

Reply:
it's ok, let's keep it.

Feel free to merge after aprove